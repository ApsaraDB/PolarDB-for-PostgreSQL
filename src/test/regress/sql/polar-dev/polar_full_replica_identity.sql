

CREATE SCHEMA IF NOT EXISTS polar_replica;
set search_path=polar_replica,sys;

alter system set polar_create_table_with_full_replica_identity = off;
select pg_reload_conf();

select pg_sleep(2);

show polar_create_table_with_full_replica_identity;

create table t_a(a int, b int);
create table t_b(a int primary key, b int);

\d+ t_a
\d+ t_b

alter system set polar_create_table_with_full_replica_identity = on;
select pg_reload_conf();

select pg_sleep(2);

show polar_create_table_with_full_replica_identity;

create table t_c(a int, b int);
create table t_d(a int primary key, b int);

\d+ t_c
\d+ t_d

alter system reset polar_create_table_with_full_replica_identity;
select pg_reload_conf();

reset search_path;
drop schema polar_replica cascade;
