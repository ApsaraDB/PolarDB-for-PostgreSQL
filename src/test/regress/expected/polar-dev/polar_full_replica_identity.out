CREATE SCHEMA IF NOT EXISTS polar_replica;
set search_path=polar_replica,sys;
alter system set polar_create_table_with_full_replica_identity = off;
select pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

select pg_sleep(2);
 pg_sleep 
----------
 
(1 row)

show polar_create_table_with_full_replica_identity;
 polar_create_table_with_full_replica_identity 
-----------------------------------------------
 off
(1 row)

create table t_a(a int, b int);
create table t_b(a int primary key, b int);
\d+ t_a
                                Table "polar_replica.t_a"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 a      | integer |           |          |         | plain   |              | 
 b      | integer |           |          |         | plain   |              | 

\d+ t_b
                                Table "polar_replica.t_b"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 a      | integer |           | not null |         | plain   |              | 
 b      | integer |           |          |         | plain   |              | 
Indexes:
    "t_b_pkey" PRIMARY KEY, btree (a)

alter system set polar_create_table_with_full_replica_identity = on;
select pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

select pg_sleep(2);
 pg_sleep 
----------
 
(1 row)

show polar_create_table_with_full_replica_identity;
 polar_create_table_with_full_replica_identity 
-----------------------------------------------
 on
(1 row)

create table t_c(a int, b int);
create table t_d(a int primary key, b int);
\d+ t_c
                                Table "polar_replica.t_c"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 a      | integer |           |          |         | plain   |              | 
 b      | integer |           |          |         | plain   |              | 
Replica Identity: FULL

\d+ t_d
                                Table "polar_replica.t_d"
 Column |  Type   | Collation | Nullable | Default | Storage | Stats target | Description 
--------+---------+-----------+----------+---------+---------+--------------+-------------
 a      | integer |           | not null |         | plain   |              | 
 b      | integer |           |          |         | plain   |              | 
Indexes:
    "t_d_pkey" PRIMARY KEY, btree (a)
Replica Identity: FULL

alter system reset polar_create_table_with_full_replica_identity;
select pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

reset search_path;
drop schema polar_replica cascade;
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to table polar_replica.t_a
drop cascades to table polar_replica.t_b
drop cascades to table polar_replica.t_c
drop cascades to table polar_replica.t_d
